.github/workflows/e2e-ci.yaml

```diff
@@ -182,6 +183,37 @@ jobs:
       env:
         CONTAINER_REGISTRY_ADDRESS: ${{ env.CONTAINER_REGISTRY_ADDRESS }}
 
+  # Generate root and server certificates and copy required files to RPi
+  certificates_job :
+    timeout-minutes: 5
+    name: Generate and copy certificates
+    environment:
+      name: CI_AZURE_ENVIRONMENT
+    if: needs.env_var.outputs.RunE2ETestsOnly != 'true' && needs.env_var.outputs.StopFullCi != 'true'
+    needs:
+      - env_var
+    runs-on: [ self-hosted, x64 ]
+
+    steps:
+    - uses: actions/checkout@v2
+      name: Checkout current branch
+
+    - id: generate_step
+      shell: bash
+      name: Generate server and client certificates
+      env:
+        SERVER_PFX_PASSWORD: ${{ secrets.SERVER_PFX_PASSWORD }}
+        SSH_PRIVATE_KEY_PATH: ${{ secrets.SSH_PRIVATE_KEY_PATH }}
+        RPI_USERNAME: ${{ secrets.RPI_USERNAME }}
+        RPI_HOST: ${{ secrets.RPI_HOST }}
+        CERT_REMOTE_PATH: ${{ needs.env_var.outputs.CertRemotePath }}
+      run: |
+        chmod +x ./Tools/BasicStation-Certificates-Generation/certificate-generate.sh
```

Comment:
QQ can't we commit it directly with the chmod +x instead of running it here?

Reply:
Yes we could, but apparently modifying it on windows (as all our dev machines) could make the executable bit disappear. Therefore, I will just leave the chmod +x here